front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-console */
+import { useState, type ReactNode } from 'react';
+
+import { DatePicker, Input } from '@osrd-project/ui-core';
+import { Gear } from '@osrd-project/ui-icons';
+import { useTranslation } from 'react-i18next';
+
+import useLinkedPathSearch from 'applications/stdcm/hooks/useLinkedPathSearch';
+import type { ExtremityPathStepType } from 'common/types';
+
+import StdcmCard from './StdcmCard';
+import StdcmDefaultCard from './StdcmDefaultCard';
+import StdcmLinkedPathResults from './StdcmLinkedPathResults';
+
+type StdcmLinkedPathSearchProps = {
+  cardIcon: ReactNode;
+  cardName: string;
+  className?: string;
+  defaultCardText: string;
+  linkedOp: ExtremityPathStepType;
+};
+
+const StdcmLinkedPathSearch = ({
+  cardIcon,
+  cardName,
+  className = '',
+  defaultCardText,
+  linkedOp,
+}: StdcmLinkedPathSearchProps) => {
+  const { t } = useTranslation('stdcm');
+  const [displayLinkedPathSearch, setShowLinkedPathSearch] = useState<boolean>(false);
+
+  const {
+    displaySearchButton,
+    launchTrainScheduleSearch,
+    linkedPathDate,
+    linkedPathID,
+    linkedPathResults,
+    selectableSlot,
+    setDisplaySearchButton,
+    setLinkedPathDate,
+    setLinkedPathID,
+    updateLinkedPathStep,
+  } = useLinkedPathSearch();
+
+  return (
+    <div className={`stdcm-linked-path-search-container ${className}`}>
+      {!displayLinkedPathSearch ? (
+        <button type="button" onClick={() => setShowLinkedPathSearch(true)}>
+          <StdcmDefaultCard text={defaultCardText} Icon={cardIcon} className="add-linked-path" />
+        </button>
+      ) : (
+        <StdcmCard
+          name={cardName}
+          title={
+            <button type="button" onClick={() => setShowLinkedPathSearch(false)}>
+              {t('translation:common.delete').toLowerCase()}
+            </button>
+          }
+          className="linked-path"
+        >
+          <div className="d-flex flex-column">
+            <div className="d-flex pr-1 pl-3">
+              <Input
+                id="linked-path-ID"
+                type="text"
+                value={linkedPathID || ''}
```

Comment:
If linkedPathID (or trainNameInput) has been initialized, you can remove the `|| ''`
```suggestion
                value={linkedPathID}
```